### PR TITLE
Fix Issue #41981"Make struct fields writable" needs to list the actual reason

### PR DESCRIPTION
**Pull request by Sliptory**

### CODE REVIEW

File: src/EditorFeatures/CSharpTest/InitializeParameter/AddParameterCheckTests.cs [hunk 1]
```diff
@@ -330,17 +330,17 @@ public C(string a, string b, string c)
     {
         if (string.IsNullOrEmpty(a))
         {
-            throw new ArgumentException(""message"", nameof(a));
+            throw new ArgumentException($""'{nameof(a)}' cannot be null or empty"", nameof(a));
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/InitializeParameter/AddParameterCheckTests.cs]: Why is this feature being changed?

File: src/EditorFeatures/Test/CodeGeneration/ExpressionGenerationTests.cs [hunk 2]
```diff
@@ -319,34 +318,6 @@ public void TestConditionalExpression1()
...
-                f => f.ConditionalAccessExpression(
-                    f.IdentifierName("E"),
-                    f.ElementBindingExpression(
-                        f.Argument(f.IdentifierName("T")))),
-                cs: "E?[T]",
-                csSimple: "E?[T]",
-                vb: "E?(T)",
-                vbSimple: "E?(T)");
-        }
-
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Test/CodeGeneration/ExpressionGenerationTests.cs]: Why was this deleted?

File: src/Features/CSharp/Portable/MakeStructFieldsWritable/CSharpMakeStructFieldsWritableDiagnosticAnalyzer.cs [hunk 3]
```diff
@@ -16,7 +16,7 @@ internal sealed class CSharpMakeStructFieldsWritableDiagnosticAnalyzer : Abstrac
         private static readonly DiagnosticDescriptor s_diagnosticDescriptor = CreateDescriptor(
             IDEDiagnosticIds.MakeStructFieldsWritable,
             new LocalizableResourceString(nameof(FeaturesResources.Make_readonly_fields_writable), FeaturesResources.ResourceManager, typeof(FeaturesResources)),
-            new LocalizableResourceString(nameof(FeaturesResources.Make_readonly_fields_writable), FeaturesResources.ResourceManager, typeof(FeaturesResources)),
+            new LocalizableResourceString(nameof(FeaturesResources.Struct_contains_assigment_to_comma_this_comma_outside_the_scope_constructor_Make_readonly_fields_writable), FeaturesResources.ResourceManager, typeof(FeaturesResources)),
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/MakeStructFieldsWritable/CSharpMakeStructFieldsWritableDiagnosticAnalyzer.cs]: Can you give a screen shot of what this looks like to the user?

File: src/Features/Core/Portable/FeaturesResources.resx [hunk 4]
```diff
@@ -1756,4 +1756,7 @@ This version used in: {2}</value>
   <data name="Implement_explicitly" xml:space="preserve">
     <value>Implement explicitly</value>
   </data>
+  <data name="Struct_contains_assigment_to_comma_this_comma_outside_the_scope_constructor_Make_readonly_fields_writable" xml:space="preserve">
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/FeaturesResources.resx]: Remove the _comma_ parts (we don't include punctuation, and those aren't commas either)

File: src/Features/CSharp/Portable/MakeStructFieldsWritable/CSharpMakeStructFieldsWritableDiagnosticAnalyzer.cs (hunk 3 above)
Comment (Sliptory) [reply src/Features/CSharp/Portable/MakeStructFieldsWritable/CSharpMakeStructFieldsWritableDiagnosticAnalyzer.cs]: answerTask

File: src/EditorFeatures/Test/CodeGeneration/ExpressionGenerationTests.cs (hunk 2 above)
Comment (Sliptory) [reply src/EditorFeatures/Test/CodeGeneration/ExpressionGenerationTests.cs]: result of merge

File: src/EditorFeatures/CSharpTest/InitializeParameter/AddParameterCheckTests.cs (hunk 1 above)
Comment (Sliptory) [reply src/EditorFeatures/CSharpTest/InitializeParameter/AddParameterCheckTests.cs]: it got here by mistake, it's a commit PR #42006

File: src/Features/Core/Portable/FeaturesResources.resx (hunk 4 above)
Comment (Sliptory) [reply src/Features/Core/Portable/FeaturesResources.resx]: ok

**Comment by CyrusNajmabadi** [comment]
@sliptory can you fixup the pr to not contain the unnecessary changes?
